PWGJE/EMCALJetTasks/macros/AddTaskJetCoreEmcal.C

```diff
@@ -31,32 +33,69 @@ AliAnalysisTaskJetCoreEmcal* AddTaskJetCoreEmcal(
   TString wagonName2 = Form("JetCore_%s_TCTree",njetsBase);
   //Configure jet tagger task
   AliAnalysisTaskJetCoreEmcal *task = new AliAnalysisTaskJetCoreEmcal(wagonName1.Data());
+	task->SetJetContName(njetsBase);
+	if(jetShapeType == AliAnalysisTaskJetCoreEmcal::kDetEmbPart) {
+		task->SetJetContPartName(njetsPart);
+		task->SetJetContTrueName(njetsTrue);
+	}
+	task->SetJetShapeType(jetShapeType);
 	task->SetTTLowRef(kTTminr);
 	task->SetTTUpRef(kTTmaxr);
 	task->SetTTLowSig(kTTmins);
 	task->SetTTUpSig(kTTmaxs);
 
 	TString name = "JetCore";
-	TString trackName = "tracks";
 	TString clusName = "caloClusters";
-  if (trackName == "mcparticles") {
-    task->AddMCParticleContainer(trackName);
-  }
-  else if (trackName == "tracks" || trackName == "Tracks") {
-    task->AddTrackContainer(trackName);
-  }
-  else if (!trackName.IsNull()) {
-    task->AddParticleContainer(trackName);
-  }
+
+	AliParticleContainer *trackCont = 0x0;
+	AliParticleContainer *trackContPartLevel = 0x0;
+	AliParticleContainer *trackContTrueLevel = 0x0;
+
+	if(jetShapeType == AliAnalysisTaskJetCoreEmcal::kMCTrue) {
+		trackCont = task->AddMCParticleContainer("mcparticles");
+	}
+	else if (jetShapeType == AliAnalysisTaskJetCoreEmcal::kData) {
+		trackCont = task->AddTrackContainer("tracks");
+	}
+	else if (jetShapeType == AliAnalysisTaskJetCoreEmcal::kDetEmbPart) {
```

Comment:
If you're intending to use this with the new embedding framework, you'll need to `SetIsEmbedding(true)` on at least some of these containers

Reply:
I was doing this in my tests by getting the relevant particle container from the task after definition, then calling SetIsEmbedding (I guess this could be done in the macro customisation as well) - would this be ok?

Reply:
Yeah, that's totally fine - I only briefly glanced at your code and didn't see it around where you were doing your configuration, so I just wanted to make sure that it wasn't lost!

Reply:
ok all good, thanks for the thought!